libraries/AP_InertialSensor/AP_InertialSensor.cpp

```diff
@@ -2240,12 +2193,96 @@ bool AP_InertialSensor::get_primary_accel_cal_sample_avg(uint8_t sample_num, Vec
     return true;
 }
 
+#if HAL_GCS_ENABLED
+bool AP_InertialSensor::calibrate_gyros()
+{
+    init_gyro();
+    if (!gyro_calibrated_ok_all()) {
+        return false;
+    }
+    AP::ahrs().reset_gyro_drift();
+    return true;
+}
+
+/*
+  calculate the trim_roll and trim_pitch. This is used for redoing the
+  trim without needing a full accel cal
+ */
+MAV_RESULT AP_InertialSensor::calibrate_trim()
+{
+    // exit immediately if calibration is already in progress
+    if (calibrating()) {
+        return MAV_RESULT_TEMPORARILY_REJECTED;
+    }
+
+    // reject any time we've done a calibration recently
+    const uint32_t now = AP_HAL::millis();
+    if ((now - last_accel_cal_ms) < 5000) {
```

Comment:
last_accel_cal_ms isn't set till end, after a bunch of waits, which seems to leave open the race. maybe set it when we start?

Reply:
This was intentional, but it may be missing some useful cases. Because we set the calibrating flag, which leads to doing a temporary rejection at the start we've mostly dealt with this. To actually benefit from using the rate limiting via `last_accel_cal_ms` we would have to use a timeout to cover the worst case scenario, which is much shorter then the current 5 second limit.